dataprofiler/tests/labelers/test_base_model.py

```diff
@@ -96,25 +96,99 @@ def test_set_parameters(self):
         with self.assertRaisesRegex(ValueError, 'test'):
             base_model.BaseModel.set_params(mock_model, **params)
 
+    def test_add_labels(self):
+
+        # setup mock
+        mock_model = mock.Mock(spec=base_model.BaseModel)
+        mock_model._convert_labels_to_label_mapping.side_effect = \
+            base_model.BaseModel._convert_labels_to_label_mapping
+        mock_model._label_mapping = {}
+        type(mock_model).num_labels = mock.PropertyMock(
+            side_effect=lambda: max(mock_model._label_mapping.values()) + 1)
```

Comment:
Shouldn't the +1 be determined by whether or not they place a label at 0? It seems like this property is dependent on the requires_zero_mapping flag

Reply:
the `requires_zero_mapping` does not impact the +1 as it is only dependent on the final value of the index. i.e. when we add a new label, it should be +1 the max value in the encodings, which is not relative but not dependent on `requires_zero_mapping`

Reply:
Okay, I guess I interpreted the name "num_labels" to be more a sign of the number of labels present, not the next free label number

Reply:
I see what you mean, this is a fair point. I'm going to update the add_label to just directly use label_mapping and ignore num_labels. Thought currently this is how the base model determines num_labels, which might be a bug we want to address.